AppWidget/README.md

```diff
@@ -0,0 +1,331 @@
+
+AppWidget Sample
+===================================
+
+This sample demonstrates how to use the new Widgets APIs introduced in API level 31.
+The new set of APIs allows an application to build more engaging and beautiful
+widgets.
+The initial set of new APIs focuses on improving existing RemoteView APIs.
+
+List of API changes
+------------
+
+- **Widget description**
+
+  In API level 31, a widget description is shown for each widget in the widget
+  picker if `description` attribute is provided for your `appwidget-provider`
+
+  ```xml
+  <appwidget-provider
+        android:description="@string/app_widget_grocery_list_description"
+    ... />
+  ```
+
+- **Preview layout**
+
+  The widget preview is displayed from the layout XML if it's provided as
+  `previewLayout` attribute for your `appwidget-provider` instead of a
+  static drawable resource using `previewImage` attribute.
+  Note that the preview may not reflect the widget after it's actually
+  placed (e.g. colors from themes)
+
+  ```xml
+    <appwidget-provider
+        android:previewLayout="@layout/widget_grocery_list"
+    ... />
+  ```
+
+  <img src="screenshots/widget_preview.png"
+    alt="Image of the widget with description and previewLayout"
+    title="Image of the widget with description and previewLayout"
+    />
+
+- **Dynamic coloring**
+
+  In API level 31, the colors of your widget will be dynamically determined by the system colors.
+  You should use the system's default theme in order to apply the dynamic coloring (`Theme.DeviceDefault` and
+  `Theme.DeviceDefault.DayNight` for v31.
+  into account).
+  ```xml
+  values/themes.xml
+
+  <style name="Theme.AppWidget.AppWidgetContainer"
+      parent="@android:style/Theme.DeviceDefault" />
+  ```
+
+  ```xml
+  values-v31/themes.xml
+
+  <style name="Theme.AppWidget.AppWidgetContainer"
+      parent="@android:style/Theme.DeviceDefault.DayNight" />
+  ```
+
+  ```xml
+  values-night-v31/themes.xml
+  <!-- Having themes.xml for night-v31 because of the priority order of the resource qualifiers. -->
+  <style name="Theme.AppWidget.AppWidgetContainer"
+      parent="@android:style/Theme.DeviceDefault.DayNight" />
+  ```
+
+  ```xml
+  layout/widget_layout.xml
+
+  <LinearLayout
+      android:theme="@style/Theme.AppWidget.AppWidgetContainer">
+    ...
+  </LinearLayout>
+  ```
+
+  <img src="screenshots/dynamic_coloring_light_theme.png" width="240px"
+      alt="screenshot for dynamic coloring on light theme"
+      title="screenshot for dynamic coloring on light theme" />
+  <img src="screenshots/dynamic_coloring_dark_theme.png" width="240px"
+      alt="screenshot for dynamic coloring on dark theme"
+      title="screenshot for dynamic coloring on dark theme" />
+
+- **Padding and rounded corners**
+
+  Following system attributes are introduced in API level 31 for widget paddings and
```

Comment:
Could we please delete this since it is no longer relevant?

Reply:
Good point. Removed the part mentioning about padding attribute. (attributes for rounded corners are still valid)